### ISSUE TITLE
remove scale and background from structure factor (Trac #657)

### ISSUE THREAD

**Issue by pkienzle** [SasView/sasview#788]
sasmodels automatically adds scale and background to every model, including structure factors.  This is built into the calculation engine.

Easiest way to implement this is to update modelinfo.py so that when self.structure_factor is True, get_hidden_parameters() hides scale and background and make_model_info() so that background is 0 (that is, parameters.kernel_parameters[1].default=0)

Migrated from http://trac.sasview.org/ticket/657
```json
{
    "status": "closed",
    "changetime": "2016-11-06T10:30:36",
    "_ts": "2016-11-06 10:30:36.177396+00:00",
    "description": "sasmodels automatically adds scale and background to every model, including structure factors.  This is built into the calculation engine.\n\nEasiest way to implement this is to update modelinfo.py so that when self.structure_factor is True, get_hidden_parameters() hides scale and background and make_model_info() so that background is 0 (that is, parameters.kernel_parameters[1].default=0)",
    "reporter": "pkienzle",
    "cc": "",
    "resolution": "fixed",
    "workpackage": "SasView Bug Fixing",
    "time": "2016-09-10T15:28:01",
    "component": "SasView",
    "summary": "remove scale and background from structure factor",
    "priority": "blocker",
    "keywords": "",
    "milestone": "SasView 4.1.0",
    "owner": "ajj",
    "type": "defect"
}
```

**Comment by butlerpd** [comment]
Trac update at `2016/09/13 14:26:21`: **butler** changed milestone from "SasView 4.0.0" to "SasView 4.1.0"

**Comment by mdoucet** [comment]
Trac update at `2016/10/08 19:50:29`: 
* **mathieu** changed owner from "" to "mathieu"
* **mathieu** changed status from "new" to "accepted"

**Comment by ajj** [comment]
Trac update at `2016/10/08 19:52:30`: **ajj** changed _comment0 from "The team decided to prepend [plug-in] to all plug-in model names." to "1476113743351627"

**Comment by mdoucet** [comment]
Trac update at `2016/10/08 19:52:30`: **mathieu** commented:

The team decided to prepend [plug-in] to all plug-in model names.

AJJ (2016-10-10) - Is this related to this ticket?

**Comment by pkienzle** [comment]
Trac update at `2016/10/10 15:21:16`: 
* **pkienzle** commented:

Changed status to blocker in accordance to Andrew's duplicate ticket.


* **pkienzle** changed priority from "minor" to "blocker"

**Comment by mdoucet** [comment]
Trac update at `2016/10/10 16:15:57`: **mathieu** commented:

Whooops. Typed in the wrong window...

**Comment by mdoucet** [comment]
Trac update at `2016/10/10 16:16:12`: 
* **mathieu** changed owner from "mathieu" to "ajj"
* **mathieu** changed status from "accepted" to "assigned"

**Comment by pkienzle** [comment]
Trac update at `2016/10/10 16:43:54`: **pkienzle** commented:

Note: the parameters are only hidden in the sasview model wrapper, not in the underlying model.  This is either a bug or a feature.  If no other applications that use sasmodels need access to background and scale, then each of them will need their own code to hide scale and parameter, so this is a bug, but if any application might want access to these parameters then it is a feature.

**Comment by ajj** [comment]
Trac update at `2016/10/10 18:21:00`: **ajj** commented:

Eventually sasmodels should not be adding scale and background for structure factors at all as this is not physically meaningful. However, that is a bigger task and the immediate issue of making sasview behave the way users would expect is fixed.

**Comment by ajj** [comment]
Trac update at `2016/10/10 18:23:01`: **ajj** commented:

In changeset 68669da9154d53fe949cb741efa850cd516ce157:

```text
#!CommitTicketReference repository="sasview" revision="68669da9154d53fe949cb741efa850cd516ce157"
Updating MultiplicationModel to deal with results of fixing #788.

Fixes #866 (again)
```

**Comment by ajj** [comment]
Trac update at `2016/11/03 09:33:19`: **ajj** commented:

In changeset 736f27e9e612490fce2428fe33d9f38530134575:

```text
#!CommitTicketReference repository="sasview" revision="736f27e9e612490fce2428fe33d9f38530134575"
Updating MultiplicationModel to deal with results of fixing #788.

Fixes #866 (again)
```

**Comment by ajj** [comment]
Trac update at `2016/11/06 10:30:36`: **ajj** commented:

In changeset be86916245e571a44945c07b0cf0f8c35364fd03:

```text
#!CommitTicketReference repository="sasmodels" revision="be86916245e571a44945c07b0cf0f8c35364fd03"
hide structure factor background/scale from sasview gui.  Fixes #788.
```
